lambdaclass/mirra_backend
# Rewrite Skills (Add Mechanics)

Closes https://github.com/lambdaclass/champions_of_mirra/issues/358

This PR adds Mechanics, which are intermediate structures between Skills and Effects. These allow for more interesting mechanics like hitting an enemy with two different effects over time with different durations, or hitting two groups of targets with many effects.

- New file nesting due to new "Mechanic" module being between Skills and Effects.
- pending_executions no longer server a purpose so they've been removed
- Fully remade the skill config to account for these changes. While we were at it, we expressed times in milliseconds. This required a few small changes in the simulator, where we translate these milliseconds into steps in the `create_x_map`.

# To test

Play some battles through the client. Everything should be fine. Battles will be slower ([this PR](https://github.com/lambdaclass/champions_of_mirra/pull/375) fixes that). Mess around with some skill configs and check everything still works :P
